### PR TITLE
[http-lib] replace httparty/httmultiparty with rest-client [backup-2.0] adapt to the new 2.0 api for backups

### PR DESCRIPTION
**Pull request by MaximilianMeister**
~~here I will try to keep track of the API changes to integrate the new API into crowbarctl in parallel. this will be necessary for writing/adapting cct tests, and testing ingeneral~~

~~so once we agreed on the location/responsibility of an API endpoint, we can immediately integrate it into crowbarctl~~

cc @vmoravec

EDIT: I have changed the context of the PR

### CODE REVIEW

File: config/crowbarrc [hunk 1]
```diff
@@ -3,3 +3,4 @@ server = http://localhost:80
 username = crowbar
 password = crowbar
 anonymous = false
+apiversion = 1.0
```
Comment (rsalevsky) [inline config/crowbarrc]: How should we do the switch to 2.0 in Cloud 7? This needs to be done somehow automatically.

**Comment by rsalevsky** [comment]
+1

**Comment by Itxaka** [comment]
Isnt this changing the base lib for requests and also changing the url routes for the backups part?

Feels like if this is the case it should be split into 2 different commits for cleanliness sake.

**Comment by MaximilianMeister** [comment]
> Isnt this changing the base lib for requests and also changing the url routes for the backups part? 

yes

> Feels like if this is the case it should be split into 2 different commits for cleanliness sake.

could you have another look? it is already done like that...

base lib: https://github.com/crowbar/crowbar-client/pull/92/commits/8939d11453906998ce72349d05a7c50bf1570f35

backup urls: https://github.com/crowbar/crowbar-client/pull/92/commits/d6a8fe9a25a19e4b2925f4db172f0e883806c1c6

**Comment by Itxaka** [comment]
@MaximilianMeister Sorry, I meant PR not commits!

Either way this looks good to me +1

**Pull request merged**